libraries/abstractions/transport/utest/CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+    project ("transportsecure sockets unit test")
+    cmake_minimum_required (VERSION 3.13)
+
+# ====================  Define your project name (edit) ========================
+    set(project_name "transport_secure_sockets")
+
+# =====================  Create your mock here  (edit)  ========================
+
+# list the files to mock here
+    list(APPEND mock_list
+                "${kernel_dir}/include/task.h"
+                "${kernel_dir}/include/portable.h"
+                "${common_dir}/include/iot_logging_task.h"
+                "${abstraction_dir}/secure_sockets/include/iot_secure_sockets.h"
+            )
+# list the directories your mocks need
+    list(APPEND mock_include_list
+                "${CMAKE_CURRENT_LIST_DIR}/include"
+            )
+#list the definitions of your mocks to control what to be included
+    list(APPEND mock_define_list
+                LWIP_COMPAT_SOCKETS=1
+                portHAS_STACK_OVERFLOW_CHECKING=1
+                portUSING_MPU_WRAPPERS=1
+                MPU_WRAPPERS_INCLUDED_FROM_API_FILE
+           )
+
+# ================= Create the library under test here (edit) ==================
+
+# The single source of truth for the transport_interface.h is in the coreMQTT repository.
+# In the FreeRTOS Device Software download page in AWS IoT Core, it is possible
+# to choose coreHTTP instead of coreMQTT. In this case, it is necessary to check
+# for the existence of the file in the downloaded software package.
+if( EXISTS ${AFR_MODULES_DIR}/coreMQTT/source )
+    set(transport_interface_dir "${AFR_MODULES_DIR}/coreMQTT/source/portable")
+elseif( EXISTS ${AFR_MODULES_DIR}/coreHTTP/source )
+    set(transport_interface_dir "${AFR_MODULES_DIR}/coreHTTP/source/portable")
+else()
+    message( FATAL_ERROR "No transport_interface.h exists for this included interface.")
+endif()
```

Comment:
Instead of repeating this logic, can we instead create a separate `transport_interface.cmake` file for placing this logic as the single source of truth? We can update the `transport/CMakeLists.txt` and `transport/utest/CMakeLists.txt` files to include the file and access the `transport_interface_dir` variable

Reply:
fixed in 388cb8a73993ee3ac039fa574771189be32f60e5